libtelemetry/src/androidTest/java/com/mapbox/android/telemetry/AlarmMangerInstrumentationTest.java

```diff
@@ -0,0 +1,60 @@
+package com.mapbox.android.telemetry;
+
+import android.app.AlarmManager;
+import android.content.Context;
+import android.content.Intent;
+import android.support.test.InstrumentationRegistry;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class AlarmMangerInstrumentationTest {
+  private int broadcastTrack;
+
+  @Test
+  public void checksAlarmCancelledProperly() {
+    broadcastTrack = 0;
+    Context context = InstrumentationRegistry.getContext();
+    AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+    AlarmReceiver alarmReceiver = obtainAlarmReceiver();
+
+    AlarmSchedulerFlusher theAlarmSchedulerFlusher = new AlarmSchedulerFlusher(context, alarmManager,
+      alarmReceiver);
+
+    long elapsedMockedTime = 2000;
+    long elapsedMockedTime2 = 5000;
+
+    theAlarmSchedulerFlusher.register();
+    theAlarmSchedulerFlusher.scheduleExact(elapsedMockedTime);
+
+    theAlarmSchedulerFlusher.register();
+    theAlarmSchedulerFlusher.scheduleExact(elapsedMockedTime2);
+
+    try {
+      Thread.sleep(30000);
```

Comment:
this doesn't look like the most efficient way of testing async code. my recommendation would be to make use of combination of counting semaphore and atomic reference. check the example of similar test here: https://github.com/mapbox/mapbox-events-android/blob/ca8b49d00d42cec753cd36cf54b04b310fe04e77/liblocation/src/test/java/com/mapbox/android/core/location/LocationEngineTest.java#L40